### PR TITLE
Guidelines for writing a connector

### LINKED ISSUE #1757: Document best practices when building connectors
### Feature description

With the many connectors we've written over time, I think we can share some good advice when it comes to writing connectors (setup, snapshots vs CDC in sources, consistent snapshots, different CDC methods, handling keys, positions, etc.).

### PR DESCRIPTION
**Pull request by hariso**
### Description

Closes #1757.

### Quick checks

- [ ] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [ ] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [ ] I have written unit tests.
- [ ] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: docs/writing-a-connector-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+in many Conduit connectors over the past few years. The document assumes basic
+knowledge about how Conduit works.
+
+## Start with the connector template
+
+The GitHub
+repository [template](https://github.com/ConduitIO/conduit-connector-template/)
+for Conduit connectors provides a basic project setup that includes placeholders
+for the source connector code, destination connector code, connector
+specifications, example of a configuration etc.
```
Comment (maha-hajja) [inline docs/writing-a-connector-guidelines.md:13]:
```suggestion
specifications, and an example of a configuration, etc.
```

File: docs/writing-a-connector-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+Also included is a `Makefile` with commonly used target, GitHub workflow for
```
Comment (maha-hajja) [inline docs/writing-a-connector-guidelines.md:15]:
```suggestion
Also included is a `Makefile` with commonly used targets. And some GitHub workflows for
```

File: docs/writing-a-connector-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+   connector.
+3. **What APIs or drivers are available?**
+
+   The choice is influenced by a number of factors, such as:
+   - Is there a driver available? If a public API and a driver are available, we
+     recommend using a driver, since it's a higher level of abstraction.
+   - In which language should the connector be written?
+   - If the language of choice is Go, is a pure Go driver available? A CGo
+     driver may make the usage and deployment of a connector more complex.
+   - Does the driver expose all the functionality needed?
```
Comment (maha-hajja) [inline docs/writing-a-connector-guidelines.md:46]:
```suggestion
   - Does the driver expose all the functionalities needed?
```

File: docs/writing-a-connector-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+   them are outlined in later sections). When researching, attention should be
+   paid if those requirements can be fulfilled.
+
+## Development
+
+## Familiarize with the Connector SDK/connector protocol
+
+Conduit's Connector SDK is the Go software development kit for implementing a
+connector for Conduit. If you want to implement a connector in another language
+please have a look at
```
Comment (maha-hajja) [inline docs/writing-a-connector-guidelines.md:73]:
```suggestion
please check
```

File: docs/writing-a-connector-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+then it's recommended to make it possible to turn off snapshots through the
+connector configuration.
+
+The following things need to be taken into account when implementing a snapshot
+procedure:
+
+- The snapshot needs to be consistent.
+- The set of the existing data can be quite large.
+- Restarting a connector during a snapshot should **not** require re-reading all
+  the data again. This is because it in some destination connectors it may cause
```
Comment (maha-hajja) [inline docs/writing-a-connector-guidelines.md:141]:
```suggestion
  the data again. This is because in some destination connectors it may cause
```

**Review by maha-hajja — COMMENTED** [review]
This document is gold! thank you for this :D

File: docs/writing-a-connector-guidelines.md [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+   the message.
+6. **Are the source/destination specific features supported?**
+
+   Source and destination connectors may have specific requirements (some of
+   them are outlined in later sections). When researching, attention should be
+   paid if those requirements can be fulfilled.
+
+## Development
+
+## Familiarize with the Connector SDK/connector protocol
```
Comment (lovromazgon) [inline docs/writing-a-connector-guidelines.md:69]: Something doesn't seem right here, I suppose the second header would need to be nested below the first?

```suggestion
## Development

### Familiarize with the Connector SDK/connector protocol
```

**Review by lovromazgon — APPROVED** [review]
👏

**Pull request merged**